CI.hs

```diff
@@ -14,9 +15,64 @@ import System.Time.Extra
 import Data.List.Extra
 import Data.Time.Clock
 import Data.Time.Calendar
+import Data.Semigroup ((<>))
+import qualified Options.Applicative as Opts
+import qualified System.Environment as Env
+import qualified System.Exit as Exit
 
 main :: IO ()
 main = do
+    let opts = Opts.info (parseOptions Opts.<**> Opts.helper)
+                 ( Opts.fullDesc
+                <> Opts.progDesc "Build (and possibly upload) ghc-lib and ghc-lib-parser tarballs."
+                <> Opts.header "CI - CI script for ghc-lib" )
+    Options upload  <- Opts.execParser opts
+    version <- buildDists
+    when upload $ bintrayUpload version
+
+newtype Options = Options Bool
```

Comment:
```suggestion
newtype Options = Options { shouldUpload :: Bool }
```

Otherwise it is very unclear what this is supposed to represent.

Reply:
Done